test/unit/manipulation.js

```diff
@@ -3099,3 +3099,71 @@ testIframe(
 		} );
 	}
 );
+
+// Custom event handler test case
+
+QUnit.test( "should handle custom '_se-custom-destroy' event correctly", function( assert ) {
+
+	// Set up the HTML structure using innerHTML within #qunit-fixture
+
+	assert.expect( 2 );
+
+	var fixture = jQuery( "#qunit-fixture" );
+
+	fixture.html( `
+    <div id="container">
+      <div class="guarded removeself" data-elt="one">
+        Guarded 1
+      </div>
+      <div class="guarded" data-elt="two">
+        Guarded 2
+      </div>
+      <div class="guarded" data-elt="three">
+        Guarded 3
+      </div>
+    </div>
+  ` );
+
+	// JavaScript code to be tested
+
+  jQuery.event.special[ "_se-custom-destroy" ] = {
+    remove: function( _handleObj ) {
+		var $t = jQuery( this );
+		console.log( $t.data( "elt" ) );
```

Comment:
Instead of patching `console.log`, use `assert.step` with `assert.verifySteps`:
https://qunitjs.com/api/assert/step/